File: UI/src/main/java/ca/warp7/rt/java/predictor/PredictorFeature.java
package ca.warp7.rt.java.predictor;

import ca.warp7.rt.java.core.ft.Feature;
import ca.warp7.rt.java.core.ft.FeatureItemTab;

import java.util.Collections;
import java.util.List;

public class PredictorFeature implements Feature {
    @Override
    public List<FeatureItemTab> getLoadedTabs() {
        return Collections.singletonList(new FeatureItemTab(
                "Match Predictor", "fas-balance-scale:16:gray", getFeatureId(),
                FeatureItemTab.Group.SingleTab, ""
        ));
    }

    @Override
    public String getFeatureId() {
        return "predictor";
    }
}

File: UI/src/main/kotlin/ca/warp7/rt/ext/predictor/PredictorFeature.kt
package ca.warp7.rt.ext.predictor

import ca.warp7.rt.context.api.Feature
import ca.warp7.rt.context.api.FeatureLink
import ca.warp7.rt.core.feature.loadParent
import ca.warp7.rt.ext.scanner.ScannerController
import javafx.scene.Parent


class PredictorFeature : Feature {
    override val link = FeatureLink("Match Predictor", "fas-balance-scale", 18)
    private var controller: PredictorController? = null


    private fun setController(controller: PredictorController) {
        this.controller = controller
    }

    override fun onClose(): Boolean {
        return true
    }

    override fun onOpen(): Pair<Parent?, Parent?> =
            null to loadParent<PredictorController>("/ca/warp7/rt/ext/predictor/predictor.fxml") {
                this.setController(it)
            }
}


